plateau/io_components/utils.py

```diff
@@ -121,6 +123,21 @@ def _ensure_compatible_indices(
     return sorted(secondary_indices)
 
 
+def group_table_by_partition_keys(table: pa.Table, partition_on: list[str]):
```

Comment:
This is the only place where polars is used. Let's figure out how to do it without it.